js/bundle.go

```diff
@@ -307,6 +310,48 @@ func (b *Bundle) instantiate(logger logrus.FieldLogger, rt *goja.Runtime, init *
 	}
 	rt.Set("__ENV", env)
 	rt.Set("__VU", vuID)
+	rt.SetPromiseRejectionTracker(func(p *goja.Promise, op goja.PromiseRejectionOperation) {
+		// TODO this only handles the case where a promise get's rejected (either by an exception or by rejecting it
+		// programmatically). But the other case where for an already rejected promise (one that we've logged the error)
```

Comment:
[HostPromiseRejectionTracker](https://tc39.es/ecma262/#sec-host-promise-rejection-tracker) which is what this sets is called in two cases:
1. a promise with no reject handler is rejected.
2. for a promise, like the one above,  a reject handler gets added.

Currently, the first one, is logged, but the idea behind the second one (as noted in the ecmascript specification) is to *not* log the occurrence of the first one as it will now be handled ;). This unfortunately means that there needs to be some logic, as I've explained below, that at least at the moment seems way too involved for me to be implemented. 

We definitely can try to do something before this is merged though